code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	effect_desc = "Draw a rune that will swap the place of the next two person that walk on the rune."
+
+obj/effect/warped_rune/bluespace
+	desc = "Everyone is everywhere at once, yet so far away from each other"
+	icon_state = "bluespace_rune"
+	runepath = /obj/effect/warped_rune/bluespace
+	var/mob/living/carbon/C // first person to run into the rune
+	var/mob/living/carbon/M //second person that run into the rune
+	var/count = 0
+
+obj/effect/warped_rune/bluespace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/swapping)
+
+
+///the first two person that stepped on the rune swap places after the second person stepped on it. TODO: fix the potential lag abyse
+obj/effect/warped_rune/bluespace/proc/swapping()
+	if(count == 0)
+		for(C in T)
+			count++
+			return
+	for(M in T)   //teleports the first person to the rune and the second person to the old location of the first person, prone to lag if abused
+		if(C == M) //the same person can't teleport with itself, it's also so people can't get rid of the "curse" by themselves
+			return
+		do_teleport(M, C, forceMove = TRUE)
+		do_teleport(C, T, forceMove = TRUE)
+		count-- //the second person becomes the first person after running into it because they teleport into the rune, triggering swapping again.
+
+
+
+
+
+
+
+
+/* basically a timestop trap, activate when ANYTHING goes over it, that includes projectiles and what not.*/
+
+
+/obj/item/slimecross/warping/sepia
+	colour = "sepia"
+	runepath = /obj/effect/warped_rune/sepiaspace
+	effect_desc = "Creates a rune that stops time for whatever pass over it."
+	drawing_time = 200 ///much longer to draw than other runes because it fucking stops time
+
+obj/effect/warped_rune/sepiaspace
+	icon_state = "time_space"
+	desc = "The clock is ticking, but in what direction?"
+	var/TS = /obj/effect/timestop///The timestop path here used during the timewarp() proc
+
+obj/effect/warped_rune/sepiaspace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/timewarp)
+
+
+///stops time on a single tile, it creates a small runtime common to all timestop.pretty minor compared to the wizard timestop.
+obj/effect/warped_rune/sepiaspace/proc/timewarp()
+	if(locate(TS) in T)//checks if there's already a timestop on the rune. here to avoid the effect triggering multiple time at the same time.
+		return
+	new TS(T, 0, 50) //make a timestop of 0 range so it only affects the tile
+
+
+
+
+
+
+
+/*Cerulean crossbreed : creates a hologram of the last person that stepped on the tile */
+
+/obj/item/slimecross/warping/cerulean
+	effect_desc = "Draw a rune that will make a hologram of the last thing that stepped on it."
+	colour = "cerulean"
+	runepath = /obj/effect/warped_rune/ceruleanspace
+
+/obj/effect/warped_rune/ceruleanspace
+	desc = "A shadow of what once passed these halls, a memory perhaps?"
+	icon_state = "holo_rune"
+	var/obj/effect/overlay/holotile
+
+
+/obj/effect/warped_rune/ceruleanspace/Initialize()
```

Comment:
used Crossed() instead of COSMIG_ATOM_ENTERED

that way, you don't have to use a for loop to go through every mob on the tile, you can just specifically copy the person who just entered the rune